src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/ConstructionTapeHelper.java

```diff
@@ -0,0 +1,289 @@
+                     {
+                         world.setBlockState(row1, Blocks.AIR.getDefaultState());
+                     }
+                    if (world.getBlockState(row2).getBlock() == ModBlocks.blockConstructionTape)
+                    {
+                        world.setBlockState(row2, Blocks.AIR.getDefaultState());
+                    }
+                }
+            }
+        }
+        else
+        {
+            for (int i = x3; i <= x1; i++)
+            {
+                for (int y = 1; y <= 255; y++)
+                {
+                    BlockPos row1 = new BlockPos(i, y, z1);
+                    BlockPos row2 = new BlockPos(i, y, z3);
+                    if (world.getBlockState(row1).getBlock() == ModBlocks.blockConstructionTape)
+                    {
+                        world.setBlockState(row1, Blocks.AIR.getDefaultState());
+                    }
+                    if (world.getBlockState(row2).getBlock() == ModBlocks.blockConstructionTape)
+                    {
+                        world.setBlockState(row2, Blocks.AIR.getDefaultState());
+                    }
+                }
+            }
+        }
+        if (z1 < z3)
+        {
+            for (int i = z1; i <= z3; i++)
+            {
+                for (int y = 1; y <= 255; y++)
+                {
+                    BlockPos row3 = new BlockPos(x1, y, i);
+                    BlockPos row4 = new BlockPos(x3, y, i);
+                    if (world.getBlockState(row3).getBlock() == ModBlocks.blockConstructionTape)
+                    {
+                        world.setBlockState(row3, Blocks.AIR.getDefaultState());
+                    }
+                    if (world.getBlockState(row4).getBlock() == ModBlocks.blockConstructionTape)
+                    {
+                        world.setBlockState(row4, Blocks.AIR.getDefaultState());
+                    }
+                }
+            }
+        }
+        else
+        {
+            for (int i = z3; i <= z1; i++)
+            {
+                for (int y = 1; y <= 255; y++)
+                {
+                    BlockPos row3 = new BlockPos(x1, y, i);
+                    BlockPos row4 = new BlockPos(x3, y, i);
+                    if (world.getBlockState(row3).getBlock() == ModBlocks.blockConstructionTape)
+                    {
+                        world.setBlockState(row3, Blocks.AIR.getDefaultState());
+                    }
+                    if (world.getBlockState(row4).getBlock() == ModBlocks.blockConstructionTape)
+                    {
+                        world.setBlockState(row4, Blocks.AIR.getDefaultState());
+                    }
+                }
+            }
+        }
+        for (int y = 1; y <= 255; y++)
+        {
+            BlockPos corner1 = new BlockPos(x1, y, z1);
+            BlockPos corner2 = new BlockPos(x1, y, z3);
+            BlockPos corner3 = new BlockPos(x3, y, z1);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'corner3' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)